pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -158,7 +163,10 @@ def on_advance_end(self):
         # -----------------------------------------
         # VALIDATE IF NEEDED + CHECKPOINT CALLBACK
         # -----------------------------------------
-        should_check_val = self._should_check_val_fx(self.iteration_count, self.is_last_batch)
+        self.progress.should_check_val = should_check_val = self._should_check_val_fx(
+            self.iteration_count, self.is_last_batch
+        )
```

Comment:
do we save the `should_check_val` to the state dict in case we resume a step in which we want to check val?

if so, doesn't the local variable `should_check_val` here get derived from the restored progress state anyway?
not sure I understand why we need to save it

Reply:
The logic to restart from it has not been added yet. We are just saving it for later PR.

Reply:
I agree with @awaelchli that ideally we don't save it but know if the batch should check val afterwards by looking at the restored state. It's maybe worth waiting to add it if not used in this PR